learning-another-language/hello-world.md

```diff
@@ -0,0 +1,279 @@
+
+![JavaScript REPL. The green "Run" button on the top is highlighted.](../assets/learning-a-new-language_hello-world_repl-run-demo.png)
+
+There are more features in replit.com that are worth exploring! For now, let's begin working on our Hello, World program.
+
+## Implementation
+
+To print `Hello, World!` in JavaScript, [we can write](https://replit.com/@adacore/hello-world#index.js):
+
+<!-- prettier-ignore-start -->
+```js
+console.log('Hello, World!');
+```
+<!-- prettier-ignore-end -->
+
+[When we run our code](https://replit.com/@adacore/hello-world#index.js), we should see `Hello, World!` in the console.
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                             |
+| --------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------- |
+| `console`                                           | A variable. The word "console" usually refers to the Terminal, command line, or wherever output logs get printed. |
+| `console.log( ... )`                                | A function call that takes in some arguments.                                                                    |
+| `'Hello, World!'`                                   | A string literal. This string is surrounded by single-quotes (`'`).                                               |
+| `;`                                                 | A semicolon.                                                                                                      |
+
+Let's debrief in slightly more detail.
+
+### The Console
+
+`console` seems a little random here. It's lowercase, like Python variables, but there seems to be no variable assignment. We don't see any `import`s or anything like that, so where did `console` come from?
+
+`console` is a variable that references a specific object: the console! This console object gives us access to wherever the program prints out output.
+
+It seems that `console` is within scope of our JavaScript file by default.
+
+At this moment, at the beginning of our JavaScript journey, we won't go deeper into this.
+
+### Calling a Function
+
+`console.log()` is calling a function. We might not know all the details about this function, but we know that it's responsible for printing objects to the console.
+
+We may make guesses that:
+
+1. The function being called here is named `log`
+1. The function `log` is a member of `console` (hence the dot-operator)
+1. We use the dot-operator to access members of objects
+1. The parens `()` is how we call and invoke the function
+
+### Single Quotes for Strings
+
+The string literal `'Hello, World!'` is surrounded by single-quotes, instead of double-quotes.
+
+If we experiment and replace it with double-quotes (`console.log("Hello, World!");`), we'll see that our code still runs as expected!
```

Comment:
expand this out into a full line rather than inline
no suggestion because code
markdown in following comment

Reply:
If we experiment and replace it with double-quotes

<!-- prettier-ignore-start -->
```js
console.log("Hello, World!");
```
<!-- prettier-ignore-end -->

we'll see that our code still runs as expected!

Reply:
hahaha thank you for the prettier ignore comments :D

Reply:
Not really necessary, right? Since prettier wants to make double quotes anyway. Oh well. 🤷